src/types.rs

```diff
@@ -55,11 +56,14 @@ impl<'a> Default for RewriteOptions<'a> {
 /// Usually the two things are too distinct to provide a common
 /// interface however for token lookup and writing back into a writer
 /// general methods are provided.
+#[non_exhaustive]
```

Comment:
Is it really worth making this non_exhaustive? Introducing new subtypes might likely require users to make changes to handle the new type. Therefore, having a major bump when introducing a new variant would be most explicit.

Reply:
I agree. It also makes dealing with this type easier downstream.